### PR TITLE
chore: improve some error messages

### PR DESCRIPTION
**Pull request by yjhmelody**

### CODE REVIEW

**Review by tadeohepperle — APPROVED** [review]
I think this is good, thanks!

File: subxt/src/error/mod.rs [hunk 1]
```diff
@@ -73,7 +73,7 @@ pub enum Error {
     Unknown(Vec<u8>),
     /// Light client error.
     #[cfg(feature = "unstable-light-client")]
-    #[error("An error occurred but it could not be decoded: {0:?}")]
+    #[error("An error occurred but it could not be decoded: {0}")]
```
Comment (lexnv) [inline subxt/src/error/mod.rs:76]: I'd be interested to know if this is something you've encountered in the wild

Comment (lexnv) [inline subxt/src/error/mod.rs:76]:
```suggestion
    #[error("Light Client error: {0}")]
```

**Review by lexnv — APPROVED** [review]
Lgtm! Thanks for contributing!

**Pull request merged**